### PR TITLE
Fix integration test case and bug in validator

### PR DESCRIPTION
**Pull request by cici37**
#### What this PR does / why we need it:
Structural schemas are considered valid even if the root object does not explicitly set Type: "Object". This wasn't noticed until the feature gate was enabled by default for CEL and k8s.io/kubernetes/test/integration/apiserver/apply: TestApplyCRDUnhandledSchema started failing consistently because it tests this specific case.
Since we have existing validation and test coverage for type required in root level:
https://github.com/kubernetes/kubernetes/blob/a4a22a25622bf426308995d99d4f708396bbde6c/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/validation.go#L121-L122
And root level type must be `object`: https://github.com/kubernetes/kubernetes/blob/a4a22a25622bf426308995d99d4f708396bbde6c/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/validation.go#L133-L135

The failing test is to test "type: array" without any items at all is completely valid: https://github.com/kubernetes/kubernetes/blob/a4a22a25622bf426308995d99d4f708396bbde6c/test/integration/apiserver/apply/apply_crd_test.go#L489-L492
Fixing the currently integration test sounds like an option. 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by cici37** [comment]
/sig api-machinery
/priority important-soon

cc [Jordan Liggitt](https://github.com/liggitt) @jpbetz

File: test/integration/apiserver/apply/apply_crd_test.go [hunk 1]
```diff
@@ -487,6 +487,7 @@ func TestApplyCRDUnhandledSchema(t *testing.T) {
 	var c apiextensionsv1beta1.CustomResourceValidation
 	err = json.Unmarshal([]byte(`{
 		"openAPIV3Schema": {
+            "type": "object",
```
Comment (liggitt) [inline test/integration/apiserver/apply/apply_crd_test.go:490]: I'm suspicious of needing to change this test case... if this was accepted previously, then shouldn't we be fixing CEL to handle this?

**Comment by cici37** [comment]
Close in favor of https://github.com/kubernetes/kubernetes/pull/1114
Note for myself: The openapi validation didn't caught the type missing and viewed the object as valid. We added the validation separately to verify items value is required when type is specified as `array` together with type missing issue which are out of the scope of the failing test.

**Comment by cici37** [comment]
The current integration test case is calling etcdclient.Put directly which bypassed the validation check which seems a little problematic.

**Comment by liggitt** [comment]
> The current integration test case is calling etcdclient.Put directly which bypassed the validation check which seems a little problematic.

That's necessary to test behavior of data which was allowed to be persisted via the v1beta1 CRD API, but can no longer be created afresh via the v1 API

**Pull request closed without merging**